aws/resource_aws_db_parameter_group.go

```diff
@@ -277,8 +277,26 @@ func resourceAwsDbParameterGroupUpdate(d *schema.ResourceData, meta interface{})
 			}
 		}
 
+		// Build a map with names from the new parameter set
```

Comment:
I would be tempted to go with:

```go
		toRemove := map[string]*rds.Parameter{}

		for _, p := range expandParameters(os.List()) {
			toRemove[*p.ParameterName] = p
		}

		for _, p := range expandParameters(ns.List()) {
			delete(toRemove, *p.ParameterName)
		}

		// Reset parameters that have been removed
		var resetParameters []*rds.Parameter
		for _, v := range toRemove {
			resetParameters = append(resetParameters, v)
		}
```

It is a bit more obvious and lets `expandParameters` care about the format of the interface it receives.

Reply:
Thanks for this writeup, @dooferlad. This does seem like an improvement. If @anGie44 agrees, I'd be happy to swap in your implementation.